### PR TITLE
Align assertion statements; change some stderr to stdout and flush

### LINKED ISSUE #321: Output errors to stderr
## (Issue Summary / Feature Request)
When unit tests run, certain information is output to stdout, and other information to stderr. It would be nice if the only information output to stderr were actual errors.

### PR DESCRIPTION
**Pull request by pianistrevor**
## Highlights from `CHANGELOG.md`

* See CHANGELOG.md for more

## Issues Fixed

* Fixes #321

### CODE REVIEW

File: cpp/unittest/ArduinoUnitTests.h [hunk 1]
```diff
@@ -76,8 +72,10 @@ class Test
           const char* description,
           bool pass
         ) {
-          cerr << "    " << (pass ? "" : "not ") << "ok " << ++mAssertCounter << " - " << description << endl;
-          if (!pass) {
+          if (pass) {
+          	cout << "    ok " << ++mAssertCounter << " - " << description << endl << flush;
```
Comment (hlovdal) [inline cpp/unittest/ArduinoUnitTests.h:76]: Indentation is off here.

File: cpp/unittest/ArduinoUnitTests.h [hunk 2]
```diff
@@ -99,9 +97,12 @@ class Test
           const char* rhsLabel,
           const B &rhs
         ) {
-          cerr << "    " << (pass ? "" : "not ") << "ok " << ++mAssertCounter << " - ";
-          cerr << description << " " << lhsLabel << " " << opLabel << " " << rhsLabel << endl;
-          if (!pass) {
+          if (pass) {
+						cout << "    ok " << ++mAssertCounter << " - ";
```
Comment (hlovdal) [inline cpp/unittest/ArduinoUnitTests.h:101]: Indentation is off here.

File: CHANGELOG.md [hunk 3]
```diff
@@ -22,6 +22,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - Update .gitattributes so we have consistent line endings
 - Change 266 files from CRLF to LF.
 - Run tests on push as well as on a pull request so developers can see impact
+- Align value indicators in assertion statements
```
Comment (hlovdal) [inline CHANGELOG.md:25]: Maybe I am being picky, but it would be great if you could split up the current commit (4bf8b53b) into two separate commits, one for each entry here.

Comment (ianfixes) [reply CHANGELOG.md:25]: Can you post a screenshot of the before/after alignment?  I'm not sure what this change is addressing

**Comment by ianfixes** [comment]
Please convert tabs to spaces in your code

**Comment by ianfixes** [comment]
Let's delay merging this until we get more discussion on #321.  I'd also like some reassurance that `cout << "blah" << flush` is functionally equivalent (in terms of segfaults) to `cerr << "blah"`.  

I don't mean to be pedantic about this, but the original code I wrote use `cout` instead of `cerr` and I made the switch after troubleshooting unit tests that had segfaults in them: https://github.com/adafruit/Adafruit-WS2801-Library/pull/20

In other words, `cerr` was the difference between seeing and not seeing the source of that segfault.

**Comment by pianistrevor** [comment]
I think I'll take just the indentation and apply suggested changes  in a new PR, while holding off on the `cerr` discussion.
